src/main/java/org/kiwiproject/dropwizard/util/startup/StartupLocker.java

```diff
@@ -0,0 +1,151 @@
+    private final SystemExecutioner executioner;
+
+    @Builder
+    private StartupLocker(SystemExecutioner executioner,
+                          ZooKeeperAvailabilityChecker zkAvailabilityChecker,
+                          CuratorFrameworkHelper curatorFrameworkHelper,
+                          CuratorLockHelper curatorLockHelper) {
+        this.executioner = Optional.ofNullable(executioner).orElseGet(SystemExecutioner::new);
+        this.zkAvailabilityChecker = Optional.ofNullable(zkAvailabilityChecker).orElseGet(ZooKeeperAvailabilityChecker::new);
+        this.curatorFrameworkHelper = Optional.ofNullable(curatorFrameworkHelper).orElseGet(CuratorFrameworkHelper::new);
+        this.curatorLockHelper = Optional.ofNullable(curatorLockHelper).orElseGet(CuratorLockHelper::new);
+    }
+
+    /**
+     * Model object containing information on the state of the acquired lock and necessary objects needed to release the lock.
+     */
+    @Getter
+    @Builder
+    public static class StartupLockInfo {
+
+        public enum LockState {
+            NOT_ATTEMPTED, ACQUIRED, ACQUIRE_FAIL
+        }
+
+        CuratorFramework client;
+        InterProcessLock lock;
+        String lockPath;
+        LockState lockState;
+        String infoMessage;
+        Exception exception;
+    }
+
+    /**
+     * Attempts to acquire a lock from ZooKeeper during startup.
+     *
+     * @param lockPath      the path in ZooKeeper to store the lock
+     * @param lockTimeout   the amount of time to wait for the lock to be acquired
+     * @param curatorConfig the Curator configuration
+     * @param environment   the Dropwizard environment
+     * @return information about the acquired lock or {@link Optional#empty()} if lock could not be acquired
+     */
+    public StartupLockInfo acquireStartupLock(String lockPath,
+                                              Duration lockTimeout,
+                                              PortAssignment assignment,
+                                              CuratorConfig curatorConfig,
+                                              Environment environment) {
+
+        if (assignment == PortAssignment.STATIC) {
+            return StartupLockInfo.builder()
+                    .lockState(StartupLockInfo.LockState.NOT_ATTEMPTED)
+                    .infoMessage("Using static port assignment. Lock not needed.")
+                    .build();
+        }
+
+        if (zkAvailabilityChecker.anyZooKeepersAvailable(curatorConfig)) {
+            var curatorFramework = curatorFrameworkHelper.startCuratorFramework(curatorConfig);
+            var lock = curatorLockHelper.createInterProcessMutex(curatorFramework, lockPath);
+
+            try {
+                tryAcquireStartupLock(lock, lockPath, lockTimeout);
+                environment.lifecycle().addLifeCycleListener(new StartupWithLockJettyLifeCycleListener(curatorFramework, lock, lockPath, executioner));
```

Comment:
Suggest putting "new StartupWithLockJettyLifeCycleListener..." on next line and adding a blank line before the `try`